### PR TITLE
chore: add contributing guide

### PR DESCRIPTION
**Pull request by johnnygreco**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+
+## Getting Started
+👋 Welcome to the Data Designer community! We're excited to have you here.
+
+Whether you're new to the project or ready to dive in, the resources below will help you get oriented and productive quickly:
+
+1. **[README.md](README.md)** – best place to start to learn the basics of the project
+
+2. **[AGENTS.md](AGENTS.md)** – code structure and development guidelines for humans and agents
```
Comment (kirit93) [inline CONTRIBUTING.md]: Can we add a bit more on what this is for?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+- Once approved, a maintainer will merge your PR
+- Your contribution will be included in the next release!
+
+## Code of Conduct
+
+Data Designer follows the Contributor Covenant Code of Conduct. We are committed to providing a welcoming and inclusive environment for all contributors.
+
+**Please read our complete [Code of Conduct](CODE_OF_CONDUCT.md)** for full details on our standards and expectations.
+
+### Copyright
```
Comment (kirit93) [inline CONTRIBUTING.md]: Where did we get this from?

File: CONTRIBUTING.md (hunk 1 above)
Comment (nabinchha) [reply CONTRIBUTING.md]: Perhaps we only say `for agents`? 
following the standards defined in https://agents.md/

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+Documentation is crucial for user adoption. Contributions that clarify usage, add examples, or fix typos are highly valued.
+
+### 💡 Examples and Tutorials
+Share your use cases! Example notebooks and tutorials help others understand how to leverage Data Designer effectively.
+
+### 🧪 Test Coverage
+Help us improve test coverage by adding tests for untested code paths or edge cases.
+
+## Feature Requests
+
```
Comment (nabinchha) [inline CONTRIBUTING.md]: white space?

File: CONTRIBUTING.md (hunk 2 above)
Comment (johnnygreco) [reply CONTRIBUTING.md]: The header requirement? OSRB

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+   make install-dev
+
+   # Or, if you use Jupyter / IPython for development
+   make install-dev-notebooks
+   ```
+
+   This creates a virtual environment in `.venv`. Activate it with:
+
+   ```bash
+   source .venv/bin/activate
```
Comment (nabinchha) [inline CONTRIBUTING.md]: is this needed if we're using uv?

File: CONTRIBUTING.md (hunk 1 above)
Comment (johnnygreco) [reply CONTRIBUTING.md]: Yeah, I was just thinking it is also a great place for humans to get a quick overview of code style / preferences / etc

I don't think we need to say much here, though. It's becoming a pretty standard.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+   ```
+
+2. **Open a pull request** on GitHub from your fork to the main repository
+
+3. **Respond to review feedback** promptly and update your PR as needed
+
+### Pull Request Review Process
+
+- Maintainers will review your PR and may request changes
+- Address feedback by pushing additional commits to your branch
```
Comment (nabinchha) [inline CONTRIBUTING.md:208]:
```suggestion
- Address feedback by pushing additional commits to your branch
- Reply to the feedback comment with a link to the commit that addresses it.
```

File: CONTRIBUTING.md (hunk 2 above)
Comment (johnnygreco) [reply CONTRIBUTING.md]: How about we change the section name to "License File Headers"?

File: CONTRIBUTING.md (hunk 1 above)
Comment (johnnygreco) [reply CONTRIBUTING.md]: What about this [ea62410](https://github.com/NVIDIA-NeMo/DataDesigner/pull/3/commits/ea6241064d46fc99a6dc3cdea35f41862598b706)

File: CONTRIBUTING.md (hunk 4 above)
Comment (johnnygreco) [reply CONTRIBUTING.md]: fair enough... old habit
